shuup/core/models/_products.py

```diff
@@ -568,7 +571,13 @@ def soft_delete(self, user=None):
             # Bypassing local `save()` on purpose.
             super(Product, self).save(update_fields=("deleted",))
 
+    # TODO (2.0): deprecate this
     def verify_mode(self):
+        warnings.warn(
+            "Product:verify_mode will be deprecated. Set the product mode explicitly.",
+            RemovedInFutureShuupWarning
+        )
```

Comment:
If you're deprecating something, it shouldn't be called in this code base anymore.

I.e. rename it to e.g. `_set_mode` and then make the deprecated `verify_mode` call `_set_mode` and other places in this class should then call `_set_mode`. If `verify_mode` is called from outside (IIRC admin calls it at least), then those places should be updated to not call it, but set the product mode explicitly (as you state in you deprecation warning).

To make sure you found all places, try raising an exception after the deprecation warning and run the tests.